### DISCUSSION TITLE
How to integrate Web3Model SDK in our own app

### DISCUSSION THREAD

**Discussion by ZeanZero** [WalletConnect/walletconnect-monorepo#4477] [Developer Support]
Hello, I encountered trouble during the integration of Web3Sdk. Please help me solve it.
In our application, we just want to use Web3Model to sign in other wallets.
The problem I have now is:
1. After calling the navController().openWeb3Modal() method, the wallet selection dialog can pop up, and the wallet can be connected under normal operation. However, when I click the disconnect button and call openWeb3Modal() again, it will still jump to the corresponding The wallet pops up and the connection is successful. At this time, it automatically jumps back to our app, which will cause our entire app to crash without any error log output.
2. Later, I wondered if there was something wrong with the way I called it. After disconnecting, the session still existed. I could get the corresponding results through the Web3Model.getAccount() and Web3Model.getSession() methods, but I still couldn’t figure it out. Know what I should do next
I think I don’t understand the entire wallet operation process, so can you tell me specifically what I should do? There are only some API calls in the official documents, but I didn't see any relevant instructions for signing through the SDK. I've been stuck for a few days. Thank you for your hard work!

**Comment by glitch-txs** [comment]
For which framework / programing language?
